pyroomacoustics/room.py

```diff
@@ -2062,18 +2002,34 @@ def compute_rir(self):
                 """
                 # fractional delay length
                 fdl = constants.get("frac_delay_length")
+
                 fdl2 = fdl // 2
 
                 # default, just in case both ism and rt are disabled (should never happen)
                 N = fdl
 
+                """
+                Source Directivity : Obj of CardioidFamily , Receiver Directivity : Obj of freq_dependent_response_sofa 
+                Source Directivity : Obj of freq_dependent_response_sofa , Receiver Directivity : Obj of CardioidFamily
+
+                Check if the directivity object of source and receiver are of same class, if not throw error
+
+                """
+
+                if self.mic_array.directivity is not None and self.sources[s].directivity is not None:
```

Comment:
I just noticed the current code restricts the source/microphone to be of the same class (CardioidFamily *or* DIRPATir).
Ideally, we would like everything to be compatible. The way to do it is to create an interface common to all types of source/mic and hide the details behind.
Let's discuss how we can do this.